File: Algorithms/Python3.x/222-Count_Complete_Tree_Nodes.py
# Runtime: 100 ms, faster than 47.09% of Python3 online submissions for Count Complete Tree Nodes.
# Memory Usage: 21.6 MB, less than 6.90% of Python3 online submissions for Count Complete Tree Nodes.
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def countNodes(self, root: TreeNode) -> int:
        if not root:
            return 0
        return (self.countNodes(root.left) if root.left is not None else 0) + 1 + (self.countNodes(root.right) if root.right is not None else 0)

File: Algorithms/C++/222-Count_Complete_Tree_Nodes.cpp
// Runtime: 40 ms, faster than 20.54% of C++ online submissions for Count Complete Tree Nodes.
// Memory Usage: 28.5 MB, less than 100.00% of C++ online submissions for Count Complete Tree Nodes.
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int countNodes(TreeNode* root) {
        if (root == NULL)
            return 0;
        int l = root->left != NULL ? countNodes(root->left) : 0;
        int r = root->right != NULL ? countNodes(root->right) : 0;
        return l + 1 + r;
    }
};
